### DISCUSSION TITLE
Add a suffix to the locale to handle multi-tenant

### DISCUSSION THREAD

**Discussion by dogfootruler-kr** [amannn/next-intl#1302] [Q&A] [answered]
Hello and first of all thank you for the library!

I have a folder structure like this:
![image](https://github.com/user-attachments/assets/256051ea-4be2-4e08-80d2-348dc41ec1c0)

And I would like to be able to call Link, redirect, useRouter function without having to set the [tenant] value all the time.

For example I want to be able to do this `redirect("/dashboard");` instead of `redirect("/tenant-name/dashboard");`

I've tried to use `createLocalizedPathnamesNavigation` with pathnames like this:
```
export const pathnames = {
  "/[slug]": {
    en: `/${env.NEXT_PUBLIC_TENANT_ID}/[slug]`,
    ko: `/${env.NEXT_PUBLIC_TENANT_ID}/[slug]`,
    ja: `/${env.NEXT_PUBLIC_TENANT_ID}/[slug]`,
    dev: `/${env.NEXT_PUBLIC_TENANT_ID}/[slug]`,
  },
} as const;
```

but sadly it does a rewrite instead of a redirect so it does not work.

Is there any way to make it work with this library?

Thank you in advance!

**Comment by amannn** [comment, marked as answer]
I guess what you could do is wrap the navigation APIs again (e.g. `LinkWithTenant`) and apply the tenant to the incoming `href` prop.

**Comment by dogfootruler-kr** [comment]
That was my second option if I couldn't do it with the library directly :).

Thank you!
